.github/workflows/release.yml

```diff
@@ -50,20 +68,125 @@ jobs:
+
+      - name: Roll over release notes
+        if: steps.configure.outputs.enabled == 'true'
+        run: |
+          cp RELEASE_NOTES.md /tmp
+          echo "" > RELEASE_NOTES.md
+
+      - name: Create release commit
+        if: steps.configure.outputs.enabled == 'true'
+        id: commit
+        run: |
+          if [[ ${{ steps.configure.outputs.release_channel }} == 'stable' ]]; then
+            git config user.name "RuffleBuild"
+            git config user.email "ruffle@ruffle.rs"
+  
+            git add -A
+            git commit -m "Release ${{ steps.version.outputs.version }}"
+          else
+            # For nightly we do not want to commit those changes,
+            # save them and then apply when building
+            git diff > /tmp/nightly-release.patch
+          fi
+
+          ref=$(git rev-parse HEAD)
+          echo "ref=$ref" | tee -a $GITHUB_OUTPUT
+
+      - name: Upload nightly release patch
+        uses: actions/upload-artifact@master
+        if: steps.configure.outputs.release_channel == 'nightly'
+        with:
+          name: nightly-release-patch
+          path: /tmp/nightly-release.patch
+
+      - name: Push release commit
+        if: steps.configure.outputs.release_channel == 'stable'
+        uses: ad-m/github-push-action@master
+        with:
+          repository: ${{ github.repository }}
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          branch: ${{ github.ref }}
+          force: false
+
       - name: Create release
-        if: steps.activity.outputs.is_active == 'true'
+        if: steps.configure.outputs.enabled == 'true'
         id: create_release
         run: |
-          tag_name="nightly-${{ steps.current_time_dashes.outputs.formattedTime }}"
-          release_name="Nightly ${{ steps.current_time_dashes.outputs.formattedTime }}"
-          gh release create "$tag_name" --title "$release_name" --generate-notes --prerelease
+          if [[ ${{ steps.configure.outputs.release_channel }} == 'nightly' ]]; then
+            tag_name="nightly-${{ steps.current_time_dashes.outputs.formattedTime }}"
+            release_name="Nightly ${{ steps.current_time_dashes.outputs.formattedTime }}"
+            notes_options="--generate-notes --prerelease"
+            package_prefix="ruffle-nightly-${{ steps.current_time_underscores.outputs.formattedTime }}"
+          else
+            tag_name="v${{ steps.version.outputs.version }}"
+            release_name="Release ${{ steps.version.outputs.version }}"
+            notes_options="--notes-file /tmp/RELEASE_NOTES.md"
+            package_prefix="ruffle-${{ steps.version.outputs.version }}"
+          fi
+
+          gh release create "$tag_name" \
+            --title "$release_name" \
+            --target "${{ steps.commit.outputs.ref }}" \
+            $notes_options
+
+          echo "tag_name=$tag_name" | tee -a $GITHUB_OUTPUT
+          echo "package_prefix=$package_prefix" | tee -a $GITHUB_OUTPUT
```

Comment:
Could we maybe extract this into a `.sh` somewhere? I'm not a fan of how long and complex our workflow is currently... and this doesn't help 😅
I'm kinda of the opinion that any non trivial block of code should be its own unit

Reply:
Moved a lot of release logic into a Python script

Reply:
Yay! Just noting that I wrote the AMO submission script in Node.JS instead of Python, because "we already use that". I too prefer Python personally, but this breaks consistency, and adds a(n admittedly ubiquitous) tooling requirement to CI.

Reply:
I'm wondering if it is really. The AMO submission is a part of an npm package the same way e.g. build.rs is a part of a crate, so it makes sense it uses the same tooling. The release script is independent from any package, and Python is available OOTB on runners.

Reply:
... personally I've been pondering on replacing some of our node scripts with rust ones 😅 

I kinda don't strongly either way and I especially don't want to put kjarosh through rewriting everything another time, but I think I'd *prefer* to not have yet another language in the mix, if we didn't have to. That said, rust is verbose for something like this, and javascript sucks.

Reply:
>  I especially don't want to put kjarosh through rewriting everything another time, but I think I'd prefer to not have yet another language in the mix, if we didn't have to. That said, rust is verbose for something like this, and javascript sucks.

Same, agreed, yup, sure. :smile: I'm fine with leaving it in Python, just wanted to mention this precedent.